# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/reporting/ReportingHeader.cpp
# Commit: af27682c5e7f
# Full Hash: af27682c5e7f86ff17d6ed152c8010e763870bdc
# Author: ssengupta <ssengupta@mozilla.com>
# Date: 2020-05-13 03:44:05
# Regressor Bug: 1635399
# File Overlap Count: 1
# Description:
#   Bug 1635399 - Function PrincipalInfoToPrincipal now returns Result<nsCOMPtr<nsIPrincipal>, nsresult> r=ckerschb
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D73868
# ==============================================================================

diff -r 86b22b1a9f7e -r af27682c5e7f dom/reporting/ReportingHeader.cpp
--- a/dom/reporting/ReportingHeader.cpp	Tue May 12 23:55:42 2020 +0300
+++ b/dom/reporting/ReportingHeader.cpp	Tue May 12 19:02:05 2020 +0000
@@ -15,6 +15,7 @@
 #include "mozilla/Services.h"
 #include "mozilla/StaticPrefs_dom.h"
 #include "mozilla/StaticPtr.h"
+#include "nsCOMPtr.h"
 #include "nsContentUtils.h"
 #include "nsIEffectiveTLDService.h"
 #include "nsIHttpChannel.h"
@@ -481,10 +482,12 @@
     const nsAString& aGroupName,
     const mozilla::ipc::PrincipalInfo& aPrincipalInfo,
     nsACString& aEndpointURI) {
-  nsCOMPtr<nsIPrincipal> principal = PrincipalInfoToPrincipal(aPrincipalInfo);
-  if (NS_WARN_IF(!principal)) {
+  auto principalOrErr = PrincipalInfoToPrincipal(aPrincipalInfo);
+  if (NS_WARN_IF(principalOrErr.isErr())) {
     return;
   }
+
+  nsCOMPtr<nsIPrincipal> principal = principalOrErr.unwrap();
   GetEndpointForReport(aGroupName, principal, aEndpointURI);
 }
 
@@ -593,13 +596,13 @@
     return;
   }
 
-  nsCOMPtr<nsIPrincipal> principal = PrincipalInfoToPrincipal(aPrincipalInfo);
-  if (NS_WARN_IF(!principal)) {
+  auto principalOrErr = PrincipalInfoToPrincipal(aPrincipalInfo);
+  if (NS_WARN_IF(principalOrErr.isErr())) {
     return;
   }
 
   nsAutoCString origin;
-  rv = principal->GetOrigin(origin);
+  rv = principalOrErr.unwrap()->GetOrigin(origin);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return;
   }